service/flags.go

```diff
@@ -71,10 +87,40 @@ func flags() *flag.FlagSet {
 		"feature-gates",
 		"Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature. '+' or no prefix will enable the feature.")
 
+	flagSet.BoolVar(&BuildFlag, buildInfoFlag, false,
+		"Displays list of components available in collector distribution in yaml format",
+	)
+
 	return flagSet
 }
 
 func getConfigFlag(flagSet *flag.FlagSet) []string {
 	cfv := flagSet.Lookup(configFlag).Value.(*configFlagValue)
 	return append(cfv.values, cfv.sets...)
 }
+
+func getBuildInfo(set CollectorSettings) error {
+	components := componentsOutput{}
+	for ext, _ := range set.Factories.Extensions {
+		components.Extensions = append(components.Extensions, ext)
+	}
+	for prs, _ := range set.Factories.Processors {
+		components.Processors = append(components.Processors, prs)
+	}
+	for rcv, _ := range set.Factories.Receivers {
+		components.Receivers = append(components.Receivers, rcv)
+	}
+	for exp, _ := range set.Factories.Exporters {
+		components.Exporters = append(components.Exporters, exp)
+	}
+	components.Version = set.BuildInfo.Version
+
+	yamlData, err := yaml.Marshal(components)
+
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(string(yamlData))
```

Comment:
considering using logger.Info instead of fmt.Println here

Reply:
Thank you so much for reviewing my PR. Please I am trying to get the output to look like this:

![image](https://user-images.githubusercontent.com/59079323/196361544-83d5bdbd-6986-45ca-863f-5ee56b9a8d0d.png)

Would logger.info, be better for this use case?

Reply:
@codeboten I think it's fine to use fmt.Println here, unless you have a specific case of logger.Info in mind where it would keep the output in a machine-readable format.

Reply:
I was thinking it would make testing easier, `fmt` is fine as well 👍🏻